server/routes/users.js

```diff
@@ -2,14 +2,272 @@
 var express = require('express');
 var router = express.Router();
 
+
+// Require database adapter file (not node-postgres directly)
+const db = require('../db')
+
 // Returns test string to verify that Users server is running.
 router.get('/ping', function(req, res, next) {
   res.send('pong - Users API');
 });
 
 // GET users listing.
 router.get('/', function(req, res, next) {
-    // TODO
+    db.query('SELECT id FROM swe_user', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//having an issue with concat
+/*
+//GET all user names.
+router.get('/', function(req, res, next) {
+    db.query('SELECT CONCAT(first_name, ' ', last_name) FROM swe_user', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+*/
+
+
+//GET all user emails.
+router.get('/emails', function(req, res, next) {
+
+    db.query('SELECT email FROM swe_user', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+ 
+//Get all user university IDs.
+router.get('/ids', function(req, res, next) {
+
+    db.query('SELECT university_id FROM swe_user;', [], (err, result) => {
+    if (err) {
+        return next(err);
+    }
+    res.send(result.rows);
+  });
+});
+
+
+//Get user info by user_id
+router.get('/<user_id>', function(req, res, next) {
+  const id = req.params.user_id;
```

Comment:
Nit: keep variable names consistent (e.g. if parameter is `user_id`, use variable `user_id`)